internal/runtime/client/client_test.go

```diff
@@ -325,6 +325,166 @@ func TestURLForExtension(t *testing.T) {
 	}
 }
 
+func Test_validateRuntimeExtensionDiscovery(t *testing.T) {
```

Comment:
```suggestion
func Test_defaultAndValidateDiscoveryResponse(t *testing.T) {
```
nit: let's please align the test func name to the func name (I think that's the standard convention and we should always to this when possible / when we are just testing the func)